### PR TITLE
corrections to beta-plane

### PR DESCRIPTION
**Pull request by raphaeldussin**
Setting up a channel with beta-plane configuration (center lat = 40N) with the following parameters resulted in abnormally large values of `Coriolis` leading to numerical instability. 

```
GRID_CONFIG = "spherical"       
SOUTHLAT = 10.0
LENLAT = 75.0
WESTLON = -200.0
LENLON = 100.0
ROTATION = "betaplane"
F_0 = 9.35E-05
BETA = 1.75E-11
```

Detail of the computation of parameters as follows:

```python
def betaplane(lat0, Rearth=6.378e6):
    """ write f0 and beta for a beta plane centered on lat0 (deg)"""
    import numpy as np
    omega = (2*np.pi/86400.) # earth rotation rate 2 pi / lenght of day
    deg2rad = 2*np.pi/360
    lat0r = deg2rad * lat0
    f0 = 2 * omega * np.sin(lat0r)
    beta = 2 * omega * np.cos(lat0r) / Rearth
    return f0, beta

f0_wide, beta_wide = betaplane(40)
print(f'wide channel: f0 = {f0_wide}, beta = {beta_wide}')
```

```
wide channel: f0 = 9.348966816711922e-05, beta = 1.7468900581468308e-11
```

I believe the y-scaling needs to be corrected according to this PR and an origin for the beta-plane is added for applications other
than the equatorial beta-plane. Experiment with updated code behaves as expected.

### CODE REVIEW

**Comment by marshallward** [comment]
Two minor comments/suggestions:
* Could the `LAT_0` parameter be moved inside the `degree` conditional, say after `RAD_EARTH`?
* Do you think renaming it to something like `BETA_LAT_REF` would be better?  We still use the flat parameter list so `LAT_0` may not be descriptive enough.

**Comment by raphaeldussin** [comment]
`LAT_0` (or `BETA_LAT_REF`) could be useful for the other conditions as well, not only the `degree` hence I've put it outside the conditional. The issue is that the units are logged in degrees when they could be any of the ones in the conditional. I can add a `get_param` after the conditional after I define a `beta_lat_ref_units` inside if that's ok with you.

I'll rename to something more self-describing.

**Comment by marshallward** [comment]
Good idea, but I think there could still be a potential mixup in the units of `geoLatBu` and `beta_lat_ref`.

You might want to use `set_grid_metrics_cartesian` as a precedent here.  It uses `AXIS_UNITS` to set the units.

**Comment by raphaeldussin** [comment]
in the `cartesian` case, geoLatBu is already in `m` or `km` since the call the `MOM_initialize_rotation` is after the `set_grid_metrics` (of which `set_grid_metrics_carthesian` is a special case) in `MOM_fixed_initialization`. However, the case for `cm` is not supported in `set_grid_metrics_carthesian` so I think we should take it off the `set_betaplane` routine.

**Comment by marshallward** [comment]
Sorry I was accidentally looking at the diff, not the file.  I missed that this function was already using `AXIS_UNITS`, so the units are probably OK.

For `cm`, I would think it's OK to just rescale `beta_lat_ref` cm to either m or km?  I agree it's a bit redundant, but also generally better to keep a feature.

**Comment by marshallward** [comment]
There was a recommendation from @Hallberg-NOAA that we retain the original answers in case they are needed for legacy experiments.

File: src/initialization/MOM_shared_initialization.F90 [hunk 1]
```diff
@@ -501,16 +503,24 @@ subroutine set_rotation_beta_plane(f, G, param_file, US)
...
       " set_rotation_beta_plane: unknown AXIS_UNITS = "//trim(axis_units))
   end select
 
+  call get_param(param_file, mdl, "BETA_LAT_REF", beta_lat_ref, &
+                 "The reference latitude (origin) of the beta-plane", &
+                 units=trim(beta_lat_ref_units), default=0.0)
+
   do I=G%IsdB,G%IedB ; do J=G%JsdB,G%JedB
-    f(I,J) = f_0 + beta * ( G%geoLatBu(I,J) * y_scl )
+    f(I,J) = f_0 + beta * ( (G%geoLatBu(I,J) - beta_lat_ref) * y_scl )
```
Comment (adcroft) [inline src/initialization/MOM_shared_initialization.F90:523]: The offset seems wrong to me: f = fo + beta * y has two degrees of freedom for a straight line but adding `beta_lat_ref` here adds up to three parameters.

**Comment by adcroft** [comment]
Which experiments are affected by the `yscl` correction?

File: src/initialization/MOM_shared_initialization.F90 (hunk 1 above)
Comment (marshallward) [reply src/initialization/MOM_shared_initialization.F90:523]: Is it not just a unit conversion?  I don't know if I'd consider that a degree of freedom.

Comment (marshallward) [reply src/initialization/MOM_shared_initialization.F90:523]: Do you mean it would be more correct to redefine `f_0` rather than introduce `beta_lat_ref`?

Comment (adcroft) [reply src/initialization/MOM_shared_initialization.F90:523]: Indeed, `f_0` can be adjusted to get the values you want. I see why `beta_lat_ref` was introduced (namely for convenience) but now there are two ways to get the same result. I guess it's just a formula with parameters, so it's not wrong. I'm not sure if too many parameters is a problem or not.

Comment (raphaeldussin) [reply src/initialization/MOM_shared_initialization.F90:523]: Imho most users expect to input parameters as in: 

https://ocw.mit.edu/courses/earth-atmospheric-and-planetary-sciences/12-808-introduction-to-observational-physical-oceanography-fall-2004/lecture-notes/course_notes_3a.pdf

With f0 being the Coriolis parameter at lat = lat0 (beta_lat_ref).

Comment (adcroft) [reply src/initialization/MOM_shared_initialization.F90:523]: I agree the generality provides the convenience to do it either way and that's probably a good thing.

**Comment by marshallward** [comment]
After some discussion, we feel reasonably confident that there is no prior work relying on this error and we can safely replace it with the correct expression.

**Review by marshallward — APPROVED** [review]

**Comment by marshallward** [comment]
https://gitlab.gfdl.noaa.gov/ogrp/MOM6/-/pipelines/12800

**Pull request merged**